alf/algorithms/sac_algorithm.py

```diff
@@ -281,8 +282,16 @@ def __init__(self,
                 "Only continuous/discrete action is supported for multidimensional reward"
             )
 
+        finite_alpha = math.isfinite(initial_log_alpha)
```

Comment:
Seems currently alpha has to be infinite to be non-trainable? But will there be cases that alpha is a plain finite number but we want to fix it unchanged?

Reply:
> Seems currently alpha has to be infinite to be non-trainable? But will there be cases that alpha is a plain finite number but we want to fix it unchanged?

If alpha is a finite number and we don't want it to be trained, then as usual we can set its learning rate as 0. This PR addresses the issue of an infinite loss computed when alpha is 0.

Reply:
How about simply giving a very negative initial_log_alpha such as -1000?

Reply:
> How about simply giving a very negative initial_log_alpha such as -1000?

But it can still be trained unless we set its learning rate to 0? I think allow an exact of 0 will be more principled.

Reply:
``infinite_alpha`` is a bit mis-leading, it is actually negtative_infinite_log_alpha?  
Feels that this newly introduced logics makes the sac algorithm a bit more complicated while only handle one additional particular case.

Reply:
> `infinite_alpha` is a bit mis-leading, it is actually negtative_infinite_log_alpha? Feels that this newly introduced logics makes the sac algorithm a bit more complicated while only handle one additional particular case.

I don't have an `infinite_alpha` variable? The `finite_alpha` is just a boolean flag. 

I didn't mean to introduce any new logic. It is just the case that our old code didn't support the case of `alpha=0`. Because in this case, the loss will become -inf and it will pollute the entire training loss. A special handling has to be needed for that.

Reply:
> > How about simply giving a very negative initial_log_alpha such as -1000?
> 
> But it can still be trained unless we set its learning rate to 0? I think allowing an exact of 0 will be more principled.

assert exp(-1000) == 0

Reply:
> > > How about simply giving a very negative initial_log_alpha such as -1000?
> > 
> > 
> > But it can still be trained unless we set its learning rate to 0? I think allowing an exact of 0 will be more principled.
> 
> assert exp(-1000) == 0

Reverted the alpha parameter change.